test/fhirtypes/ElementDefinition.assignString.test.ts

```diff
@@ -0,0 +1,1095 @@
+
+describe('ElementDefinition', () => {
+  let defs: FHIRDefinitions;
+  let observation: StructureDefinition;
+  let medication: StructureDefinition;
+  let patient: StructureDefinition;
+  let riskEvidenceSynthesis: StructureDefinition;
+  let location: StructureDefinition;
+  let capabilityStatement: StructureDefinition;
+  let imagingStudy: StructureDefinition;
+  let device: StructureDefinition;
+  let task: StructureDefinition;
+  let fisher: TestFisher;
+
+  beforeAll(() => {
+    defs = new FHIRDefinitions();
+    loadFromPath(
+      path.join(__dirname, '..', 'testhelpers', 'testdefs', 'package'),
+      'testPackage',
+      defs
+    );
+    fisher = new TestFisher().withFHIR(defs);
+  });
+  beforeEach(() => {
+    observation = fisher.fishForStructureDefinition('Observation');
+    medication = fisher.fishForStructureDefinition('Medication');
+    patient = fisher.fishForStructureDefinition('Patient');
+    riskEvidenceSynthesis = fisher.fishForStructureDefinition('RiskEvidenceSynthesis');
+    location = fisher.fishForStructureDefinition('Location');
+    capabilityStatement = fisher.fishForStructureDefinition('CapabilityStatement');
+    imagingStudy = fisher.fishForStructureDefinition('ImagingStudy');
+    device = fisher.fishForStructureDefinition('Device');
+    task = fisher.fishForStructureDefinition('Task');
+  });
+  describe('#assignString', () => {
+    // Assigning a string
+    it('should assign a string to a string', () => {
+      const batchLotNumber = medication.elements.find(e => e.id === 'Medication.batch.lotNumber');
+      batchLotNumber.assignValue('foo bar');
+      expect(batchLotNumber.patternString).toBe('foo bar');
+      expect(batchLotNumber.fixedString).toBeUndefined();
+    });
+
+    it('should assign a string to a string (exactly)', () => {
+      const batchLotNumber = medication.elements.find(e => e.id === 'Medication.batch.lotNumber');
+      batchLotNumber.assignValue('foo bar', true);
+      expect(batchLotNumber.fixedString).toBe('foo bar');
+      expect(batchLotNumber.patternString).toBeUndefined();
+    });
+
+    it('should throw ValueAlreadyAssignedError when assigning an already assigned string by pattern[x]', () => {
+      const batchLotNumber = medication.elements.find(e => e.id === 'Medication.batch.lotNumber');
+      batchLotNumber.assignValue('foo bar');
+      expect(batchLotNumber.patternString).toBe('foo bar');
+      expect(() => {
+        batchLotNumber.assignValue('bar foo');
+      }).toThrow(
+        'Cannot assign "bar foo" to this element; a different string is already assigned: "foo bar".'
+      );
+      expect(() => {
+        batchLotNumber.assignValue('bar foo', true);
+      }).toThrow(
+        'Cannot assign "bar foo" to this element; a different string is already assigned: "foo bar".'
+      );
+    });
+
+    it('should throw ValueAlreadyAssignedError when assigning an already assigned string by assigned[x]', () => {
```

Comment:
There's a handful of places in here that refer to `assigned[x]`. I think it should still be `fixed[x]` right?

Reply:
Correct!